### PR TITLE
feat: add check to prevent releases with "latest" version dependencies

### PR DESCRIPTION
**Pull request by maze88**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -408,6 +408,7 @@ bundle-push: ## Push the bundle image.
 
 .PHONY: release-build
 release-build:
+	yq '.[].version'  hack/release.yaml | grep 'latest' && exit 1
```
Comment (rollandf) [inline Makefile:411]: grep return with `1` if no matches. No need for: ` && exit 1`

Comment (maze88) [reply Makefile:411]: But if we *find* 'latest' we *want* it to fail...

Comment (rollandf) [reply Makefile:411]: right, you still need to add `|| true` at the end

`yq '.[].version' hack/release.yaml | grep -q 'latest' && exit 1 || true`

Comment (maze88) [reply Makefile]: Good catch! Fixed.

**Review by rollandf — APPROVED** [review]

**Pull request merged**